src/components/Icon.Types.ts

```diff
@@ -0,0 +1,8 @@
+export type IconPath = object | string
```

Comment:
TypeScript's `object` type has some very weird properties.

```ts
export type IconPath = object | string
const iconPath: IconPath = new Number(1)
```

Is valid, for example, because `new Number` produces an object :scream: [Here's a good StackOverflow clarifying the difference between them](https://stackoverflow.com/questions/49464634/difference-between-object-and-object-in-typescript) (which I have to reference constantly because I cannot keep them straight in my head!)

In this case we need to indicate that it can be the value of an `svg` import, right?

https://github.com/WordPress/openverse-frontend/blob/4552c37e4f8f0f2f17eb46b3d79073b39c3844bb/typings/openverse/index.d.ts#L1-L9

All `svg` imports are typed as `string` so I think just the string type should be okay. If the type definition there is wrong though, and there's something more specific, then we can change the definition to be more accurate, but I don't see much value in it as we're likely not going to, and probably shouldn't, work on those values directly.

Reply:
This was done to mimic the check for `isTest` in the Vue component props validation to make sure that tests (that for some reason receive `iconPath` as an object, not as a string) don't break. Now I see that if we declare the type in the interface as a `string`, and keep the `isTest ? object : string` check in the prop validator, it all works as it should! So, I simply left the TS type as `string`.
Thank you for the explanation! I am slowly learning about module typings as well.